XcodeServerSDK/Server Entities/BotConfiguration.swift

```diff
@@ -132,9 +132,19 @@ public class BotConfiguration : XcodeServerEntity {
         self.schedule = BotSchedule(json: json)
         self.triggers = XcodeServerArray(json.arrayForKey("triggers"))
         self.testingDestinationType = TestingDestinationIdentifier(rawValue: json.intForKey("testingDestinationType"))
-        self.testingDeviceIDs = json.arrayForKey("testingDeviceIDs")
         self.sourceControlBlueprint = SourceControlBlueprint(json: json.dictionaryForKey("sourceControlBlueprint"))
         
+        //old bots (xcode 6) only have testingDeviceIds, try to parse those into the new format of DeviceSpecification (xcode 7)
+        if let deviceSpecJSON = json.optionalDictionaryForKey("deviceSpecification") {
+            self.deviceSpecification = DeviceSpecification(json: deviceSpecJSON)
+        } else {
+            if let testingDeviceIds = json.optionalArrayForKey("testingDeviceIDs") as? [String] {
```

Comment:
Yeah, in this case I prefer the longer, but simpler to understand solution. I actually prefer that most times, because the compiler reorders/optimizes the hell out of it anyway, so we can at least write readable code :blush: But thanks for the suggestion, that's a good way to write it really concisely.